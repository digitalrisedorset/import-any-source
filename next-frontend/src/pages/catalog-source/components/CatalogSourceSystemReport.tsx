import {ReportStyles, Title} from "../../global/styles/ReportStyles";
import {CatalogSourceState} from "@/pages/types/states";

type CatalogSourceSystemReportProps = {
    currentCatalogSource: CatalogSourceState
}

export const CatalogSourceSystemReport: React.FC = ({currentCatalogSource}: CatalogSourceSystemReportProps) => {

    const ImportedMessage = () => {
        let message = ''
        if (currentCatalogSource?.numberCatalogSourceAttributes === 0) {
            message = message.concat(`no attributes were imported`)
        } else {
            message = message.concat(`${currentCatalogSource?.numberCatalogSourceAttributes} attributes were imported`)
        }

        return <p>{message}</p>
    }

    const IgnoredMessage = () => {
        if (currentCatalogSource?.numberCatalogSourceAttributes === 0) {
            return ''
        }
        let message = ''
        if (currentCatalogSource?.ignoredAttributes === 0) {
            message = message.concat(`all attributes are active`)
        } else {
            message = message.concat(`${currentCatalogSource?.ignoredAttributes} attributes are setup as ignored`)
        }

        return <p>{message}</p>
    }

    return (
        <ReportStyles>
            <div className="report">
                <Title>Catalog Source Report for {currentCatalogSource.name} attributes</Title>
                <span className="type">{ImportedMessage()}</span>
                <span className="type">{IgnoredMessage()}</span>
            </div>
        </ReportStyles>)
        }