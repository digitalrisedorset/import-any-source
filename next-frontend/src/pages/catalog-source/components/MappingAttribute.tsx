import {useRouter} from "next/router";
import {StepForm} from "../../global/styles/StepForm";
import {useMappingVerifier} from "../../mapping/hooks/useMappingVerifier";
import {MappingReport} from "./MappingReport"

export const MappingAttributes: React.FC = () => {
    const mappingReady = useMappingVerifier()

    const router = useRouter()

    async function handleSubmit(e: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
        e.preventDefault();
        router.push({ pathname: `/catalog-source`});
    }

    return (
        <StepForm>
            <div className="main">
                <h2>Step 3</h2>

                <button type="submit" disabled={!mappingReady} onClick={handleSubmit}>
                    Mapping Attributes
                </button>
            </div>
            <MappingReport/>
        </StepForm>
)
}