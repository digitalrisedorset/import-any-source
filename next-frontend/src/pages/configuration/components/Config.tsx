import {CardStyles} from "../styles/CardStyles";
import {Label} from "@/pages/global/styles/Form"
import {CatalogSource} from "../styles/CatalogSource";
import {ConfigReader} from "../models/ConfigReader";
import {useThemeSelect} from "@/pages/configuration/hooks/useThemeSelect";

export const Config = () => {
    const setCurrentTheme = useThemeSelect()
    const configReader = new ConfigReader()

    const onThemeChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
        e.preventDefault()
        setCurrentTheme(e.target.value)
    };

    return (
        <CardStyles>
            <h2>System Preferences</h2>
            <CatalogSource>
                <fieldset>
                    <Label htmlFor="code">Select your theme</Label>
                    <select onChange={onThemeChange} className="form-select">
                        <option value="">-</option>
                        {configReader.getThemeOptions().map((item) => {
                            return (<option key={item.value} value={item.value}>{item.label}</option>)
                        })}
                    </select>
                </fieldset>
            </CatalogSource>
        </CardStyles>
    )
}