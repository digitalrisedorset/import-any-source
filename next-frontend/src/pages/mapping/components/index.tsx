import { MapField } from "./MapField";
import { MatchingField } from "./MatchingField";
import React from "react";
import {useCatalogSourceMapping} from "@/state/catalogSourceMappingState";

export const Map: React.FC = () => {
    const { catalogSourceAttribute, magentoMatches } = useCatalogSourceMapping()

    return (
        <>
            <MapField />
            <MatchingField catalogSourceAttribute={catalogSourceAttribute} magentoMatches={magentoMatches} />
        </>
    )
}