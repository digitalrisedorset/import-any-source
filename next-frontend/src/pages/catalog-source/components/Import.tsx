import {ImportHome} from '../../global/styles/MappingScreen';
import {TabButtons} from "../../global/components/TabButtons";
import {TabContent} from "../../global/components/TabContent";
import {useState} from "react";
import {StepDataReader} from "../../global/models/StepDataReader";
import {TabContainer} from "../../global/styles/TabButtonsStyle";
import {useAccess} from "@/pages/user-authentication/hooks/useAccess";

export const Import: React.FC = () => {
    const stepDataReader = new StepDataReader()
    const [activeTab, setActiveTab] = useState<number>(0);
    const stepData = stepDataReader.getImportStep();
    const {canSetupImport} = useAccess()

    return (
        <ImportHome>
            {canSetupImport &&
            <TabContainer>
                <TabButtons
                    activeTab={activeTab}
                    setActiveTab={setActiveTab}
                    stepData={stepData}
                />
                <TabContent stepData={stepData} activeTab={activeTab} />
            </TabContainer>
            }
        </ImportHome>
    )
}