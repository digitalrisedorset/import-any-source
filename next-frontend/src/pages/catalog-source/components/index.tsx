import { GetCatalogSourceAttribute } from "./MainMappingArea/GetCatalogSourceAttribute";
import { MappingScreen } from '../../global/styles/MappingScreen';
import { GetIgnoredAttribute } from "./IgnoreFieldArea/GetIgnoredAttribute";
import {useFlashMessage} from "@/state/flassMessageState";
import {useCatalogSourceAttributes} from "@/pages/catalog-source/graphql/useCatalogSourceAttributes";
import React from "react";
import {LoadingDotsIcon} from "@/pages/global/components/Loading";

export const CatalogSource: React.FC = () => {
    const { initialAttribute, matchingAttribute } = [null, null]
    const { addFlashMessage } = useFlashMessage()
    const { data, error, loading } = useCatalogSourceAttributes()

    if (initialAttribute && matchingAttribute) {
        addFlashMessage(`The catalog attribute "${initialAttribute}" is matched with the magento attribute "${matchingAttribute}"`)
    }

    if (loading) return <LoadingDotsIcon />

    return (
        <MappingScreen>
            {error && <h3>{error.message}</h3>}
            <div className="fields">
                <GetCatalogSourceAttribute />
            </div>
            <GetIgnoredAttribute catalogSourceAttributes={data?.catalogSourceAttributes} />
        </MappingScreen>
    )
}