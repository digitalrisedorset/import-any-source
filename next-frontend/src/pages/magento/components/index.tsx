import {ReadMagentoAttribute} from "./ReadMagentoAttribute";
import React from "react";
import {useMagentoAttributes} from "@/pages/magento/graphql/keystone/useMagentoAttributes";
import {LoadingDotsIcon} from "@/pages/global/components/Loading";

export const Magento: React.FC = () => {
    const { data, error, loading } = useMagentoAttributes()

    if (loading) return <LoadingDotsIcon />

    return (
        <ReadMagentoAttribute magentoAttributes={data?.magentoAttributes} />
    )
}