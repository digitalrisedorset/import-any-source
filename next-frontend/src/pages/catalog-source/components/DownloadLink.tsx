import {DownlinkFile} from "@/pages/types/catalog-source";
import {DownloadLink} from "@/pages/global/styles/FlashMessage"


export const RenderFileDownload: React.FC<DownlinkFile> = (link: DownlinkFile) => (
    <DownloadLink href={link.fileurl}><span>{link.filename}</span></DownloadLink>
)