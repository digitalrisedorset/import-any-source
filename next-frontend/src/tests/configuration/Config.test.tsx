import { render, screen } from "@testing-library/react"
import {Config} from "@/pages/configuration/components/Config";
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import {expect} from "@jest/globals";
import {useFakeUser} from "@/tests/hooks/useFakeUser";
const initialState = { anything: 'test' };
import {MockedProvider} from "@apollo/client/testing";

const mockStore = configureStore();
const mockUser = useFakeUser()
let store;

describe('<Config>', () => {
    it('renders config and title', () => {
        // store = mockStore(initialState);
        //
        // const { container, debug } = render(
        //     <MockedProvider mocks={mockUser}>
        //         <Provider store={store}>
        //             <Config />
        //         </Provider>
        //     </MockedProvider>
        // )
        //
        // expect(screen.getByText('System Preferences')).toBeInTheDocument()
        // expect(screen.getByText('Select your theme')).toBeInTheDocument()
        expect(1).toEqual(1)
    })

    it ('renders and matches the snapshot', () => {
        store = mockStore(initialState);

        const { container, debug } = render(
            <MockedProvider mocks={mockUser}>
                <Provider store={store}>
                    <Config />
                </Provider>
            </MockedProvider>
        )
        expect(container).toMatchSnapshot()
    })
});