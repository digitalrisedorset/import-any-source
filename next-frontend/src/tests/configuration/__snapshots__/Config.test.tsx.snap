// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Config> renders and matches the snapshot 1`] = `
<div>
  <div
    class="sc-braxZu gWPSqz"
  >
    <h2>
      System Preferences
    </h2>
    <div
      class="sc-fQpRED eOOFKd"
    >
      <fieldset>
        <p
          class="sc-elDIKY iBAOLe"
          for="code"
        >
          Select your theme
        </p>
        <select
          class="form-select"
        >
          <option
            value=""
          >
            -
          </option>
          <option
            value="blue"
          >
            blue
          </option>
          <option
            value="red"
          >
            red
          </option>
          <option
            value="happy"
          >
            happy
          </option>
          <option
            value="forest"
          >
            forest
          </option>
        </select>
      </fieldset>
    </div>
  </div>
</div>
`;
